src/psyclone/domain/lfric/kernel/field_vector_arg.py

```diff
@@ -0,0 +1,158 @@
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford, STFC Daresbury Lab
+
+'''Module containing the FieldVector Arg class which captures the metadata
+associated with a field vector argument. Supports the creation, modification
+and Fortran output of a Field Vector argument.
+
+'''
+from psyclone.domain.lfric.kernel.field_arg import FieldArg
+
+
+class FieldVectorArg(FieldArg):
+    '''Class to capture LFRic kernel metadata information for a field
+    vector argument.
+
+    :param Optional[str] datatype: the datatype of this field \
+        (GH_INTEGER, ...).
+    :param Optional[str] access: the way the kernel accesses this \
+        field (GH_WRITE, ...).
+    :param Optional[str] function_space: the function space that this \
+        field is on (W0, ...).
+    :param Optional[str] vector_length: the size of the vector.
+
+    '''
+    def __init__(self, datatype=None, access=None, function_space=None,
+                 vector_length=None):
+        super().__init__(
+            datatype=datatype, access=access, function_space=function_space)
+
+        if vector_length is None:
+            self._vector_length = vector_length
+        else:
+            self.vector_length = vector_length
+
+    @staticmethod
+    def create_from_fortran_string(fortran_string):
+        '''Create an instance of this class from a Fortran string.
+
+        :param str fortran_string: a string containing the metadata in \
+            Fortran.
+
+        :returns: an instance of FieldVectorArg.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.FieldVectorArg`
+
+        '''
+        fparser2_tree = FieldVectorArg.create_fparser2(fortran_string)
+        return FieldVectorArg.create_from_fparser2(fparser2_tree)
+
+    @staticmethod
+    def create_from_fparser2(fparser2_tree):
+        '''Create an instance of this class from an fparser2 tree.
+
+        :param fparser2_tree: fparser2 tree capturing the metadata \
+            for a field vector argument.
+        :type fparser2_tree: :py:class:`fparser.two.Fortran2003.Part_Ref`
+
+        :returns: an instance of this class.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.FieldVectorArg`
+
+        :raises TypeError: if the first metadata argument is not in \
+            the form datatype*vector_length.
+
+        '''
+        FieldVectorArg.check_fparser2(fparser2_tree, nargs=4)
+        vector_datatype = fparser2_tree.children[1].children[0].tostr()
+        components = vector_datatype.split("*")
+        if len(components) != 2:
+            raise TypeError(
+                f"Expecting the first argument to be in the form "
+                f"'form*vector_length' but found '{vector_datatype}'.")
+        vector_length = components[1].strip()
+
+        datatype = fparser2_tree.children[1].children[1].tostr()
```

Comment:
So a FieldVectorArg would internally have a FieldArg and a vector length, rather than being a subclass of FieldArg? Anyway I'm not going to change it :-)